### PR TITLE
Add Tesseract_process_managers

### PR DESCRIPTION
**Pull request by mpowelson**
Adds process managers based on [Taskfow](https://github.com/taskflow/taskflow/blob/master/CMakeLists.txt) for orchestrating Tesseract processes. Currently these include freespace and raster path motion planning, where each individual motion plan is executing in parallel based on the required dependencies. For example, freespace motions between rasters require that each raster already be planned. Additionally, sub-Taskflows are defined to facilitate defining a "fail-active" motion planning approach where a failure tree of planners can be defined for each raster or freespace motion.

### CODE REVIEW

**Comment by mpowelson** [comment]
Some of these changes are in #332, so I will rebase as soon as that is merged.

**Comment by mpowelson** [comment]
Ok @Levi-Armstrong Kinetic is going to fail unless we add your PPA to the docker or upgrade the cmake version. I'm not sure what that Focal error is.

But this should be ready for at least an architectural look. I'm going to move on from tracking minor bugs for now and start on the  raster process

File: tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/move_instruction.h [hunk 1]
```diff
@@ -54,16 +55,17 @@ class MoveInstruction
   using Ptr = std::shared_ptr<MoveInstruction>;
   using ConstPtr = std::shared_ptr<const MoveInstruction>;
 
-  MoveInstruction(Waypoint waypoint, MoveInstructionType type, const std::string& profile = "DEFAULT");
+  MoveInstruction(Waypoint waypoint,
+                  MoveInstructionType type,
+                  const std::string& profile = "DEFAULT",
+                  ManipulatorInfo manipulator_info = ManipulatorInfo());
```
Comment (Levi-Armstrong) [inline tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/move_instruction.h:61]: Should this be a required by the user to provide? The only field that must be field out is `manipulator`

File: tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/manipulator_info.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ */
+struct ManipulatorInfo
+{
+  /** @brief Name of the manipulator group */
+  std::string manipulator;
+
+  /** @brief (Optional) IK Solver to be used */
+  std::string manipulator_ik_solver;
+
+  /** @brief The tool center point */
```
Comment (Levi-Armstrong) [inline tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/manipulator_info.h:51]: Mark this optional?

File: tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/manipulator_info.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  Eigen::Isometry3d tcp{ Eigen::Isometry3d::Identity() };
+
+  /** @brief The working frame the waypoint is relative to */
```
Comment (Levi-Armstrong) [inline tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/manipulator_info.h:54]: Mark this optional and add that if empty the base link of the environment will be used?

File: tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/move_instruction.h [hunk 4]
```diff
@@ -54,16 +55,17 @@ class MoveInstruction
...
   void setWaypoint(Waypoint waypoint);
   const Waypoint& getWaypoint() const;
 
-  void setTCP(const Eigen::Isometry3d& tcp);
-  const Eigen::Isometry3d& getTCP() const;
-
-  void setWorkingFrame(std::string working_frame);
-  const std::string& getWorkingFrame() const;
+  void setManipulatorInfo(ManipulatorInfo info);
+  const ManipulatorInfo& getManipulatorInfoConst() const;
```
Comment (Levi-Armstrong) [inline tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/move_instruction.h:67]: You can remove the `Const` from the name.

Comment (mpowelson) [reply tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/move_instruction.h:67]: When they are the same name it errors saying I am trying to overload based on return type.

File: tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/move_instruction.h (hunk 1 above)
Comment (mpowelson) [reply tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/move_instruction.h:61]: Maybe. I guess it's unlikely that things would work if you don't fill it out, but you could set it later. 

Would you rather?
```
  MoveInstruction(Waypoint waypoint,
                  MoveInstructionType type,
                 ManipulatorInfo manipulator_info
                  const std::string& profile = "DEFAULT");
```

File: tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/manipulator_info.h (hunk 3 above)
Comment (mpowelson) [reply tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/manipulator_info.h:54]: done

File: tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/move_instruction.h (hunk 4 above)
Comment (mpowelson) [reply tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/move_instruction.h:67]: done

File: tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/manipulator_info.h (hunk 2 above)
Comment (mpowelson) [reply tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/manipulator_info.h:51]: done

**Comment by mpowelson** [comment]
Added the SimpleMotionPlanner. This should fill the need for a way to initialize the seed and also acts as a planner that does interpolation depending on what functions you put in the profile.  I still need to make a generateSeed-like wrapper for it and integrate it into my process manager examples. An eagle-eyed reviewer will also notice that many of the functions I have defined are not yet implemented. These shouldn't take long to do, but I don't know if I'll get to them this week. I imagine we will also want to make one that initializes them to a value (e.g. null or the current joint value).

**Comment by mpowelson** [comment]
@Levi-Armstrong Would it be possible/advisable to make a default constructor for Instruction that constructs it as a NullInstruction? Ie so you can do things like
```
Instruction instruction;
...
instruction = CompositeInstruction()
```

**Comment by Levi-Armstrong** [comment]
I don't see an issue with that.

**Comment by mpowelson** [comment]
@Levi-Armstrong The tests are still failing. If I had to make a todo list I'd say

- [ ] Fix broken unit tests
- [ ] Implement the task validators for 
  - [ ] Format check (for checking that seed matches instruction)
  - [ ] Discrete Collision
  - [ ] Continous collision
  - [ ] etc
- [ ] Implement the rest of the step generators
- [ ] Address all of the todos placed everywhere in the code.
- [ ] Unit tests. I tried to structure everything such that components are easily testable. However, not many of them are currently tested.

**Comment by Levi-Armstrong** [comment]
continued in PR #337

**Pull request closed without merging**